cmake/CompilerOptions.cmake

```diff
@@ -61,6 +61,7 @@ elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
                  # This warning is purely informational
             4244 # Narrowing conversions: Too strict and noisy for this code base
             4242 # Narrowing conversions: Too strict and noisy for this code base
+            4820 # The type and order of elements caused the compiler to add padding to the end of a struct.
```

Comment:
It would be good if the commit message included info about what parts of the code triggered this warning.

Reply:
This is the code:

`struct RollPitchYaw
{
    RotationConvention convention;
    Eigen::Array3d rollPitchYaw;
};

struct Representations
{
    Eigen::AngleAxisd axisAngle;
    Eigen::Vector3d rotationVector;
    Eigen::Quaterniond quaternion;
    std::array<RollPitchYaw, nofRotationConventions> rotations;
};`

This is the warning:

warning C4820: 'RollPitchYaw': '4' bytes padding added after data member 'RollPitchYaw::convention'
warning C4820: 'Representations': '8' bytes padding added after data member 'Representations::rotationVector'

How does the line below sound?

The type and order of elements caused the compiler to add padding to the end of a struct; specifically: RollPitchYaw::convention and Representations::rotationVector

If we add more code that would trigger this warning we would not catch it in the future because we have already disabled the warning. Then the comments with info about what parts of the code triggered this warning would not be complete right? Is this then the best practice?

Reply:
The good thing about comments in git commits is that they are tied to a specific point in time. When I read a git commit comment, I know that it's a comment about the state of the code right then. I do not expect it to be valid in the future.

(This is opposed to comments in the source code itself, which need to be kept up to date as the code changes.)

Git commit comments are very useful when someone in the future wonders why a certain thing was done in the past. Say I want to know why this warning was ignored. Then it's very easy to find the commit which ignored it, and I should then be able to read an explanation in the commit message.

I think your suggested commit message there is good.